README.md
# OsteAPI Dokumentation

## Resurser

### GET /cheeses

Resursen repræsenterer alle oste i kartoteket
Resursen er formateret på følgende måde:

```JSON
    [{
        "varenr": 1234,
        "navn": "St clemens",
        "pris": 42.5,
        "type": "Blåskimmel",
        "producent": "St clemens",
        "antal": 15,
        "beskrivelse": {
            "tekst": "Den ost er bare super smagfuld og dejlig indeni...",
            "smag": "mild",
            "vægt": 300,
            "billede": "stclemensblaaskimmel.jpg"
        }
    }]
    
```


### GET /cheeses/\<varenummer>

Resursen repræsenterer en enkelt ost i kartoteket.
Resursen er formateret på følgende måde:

```JSON
    {
        "varenr": 1234,
        "navn": "St clemens",
        "pris": 42.5,
        "type": "Blåskimmel",
        "producent": "St clemens",
        "antal": 15,
        "beskrivelse": {
            "tekst": "Den ost er bare super smagfuld og dejlig indeni...",
            "smag": "mild",
            "vægt": 300,
            "billede": "stclemensblaaskimmel.jpg"
        }
    }
```

CHANGELOG.md
# Changelog

**1.1.0** - 20170907

Oprettet en funktion: getOne

Oprettet funktion: getAll

API dokumenteret i README.md 

Funktioner dokumenteret med JSDOC

**1.0.0** - 20170904

Server oprettet

Routes modulariseret til enkeltfiler

Routes benytter statiske objekter

services/cheeses.js
const cheeses = require ('../data/cheese.js');

/**
 * @module Cheeses
 */
module.exports = {
    /**
     * getAll
     * Returnerer et JSON object med alle oste fra kartoteket
     * @returns {Object}
     */
    'getAll': () => cheeses,

    /**
     * getOne
     * Returnerer et JSON object med en enkelt ost fra kartoteket
     * @param {string}varenr - varenummer på en ost
     * @returns {Object}
     */

    'getOne': (varenr) => {
        let currentCheese = null;
        cheeses.forEach((cheese) => {
            if (cheese.varenr == varenr) {
                currentCheese = JSON.stringify(cheese);

            }
        });

        if (currentCheese !== null) {
            return(currentCheese);
        
        }else{
            return{
                "code": "ResourceNotFound",
                "message": varenr + " does not exist"
            };
        }
    }
};

data/cheese.js

module.exports = [{
        "varenr": 1234,
        "navn": "St clemens",
        "pris": 42.5,
        "type": "Blåskimmel",
        "producent": "St clemens",
        "antal": 15,
        "beskrivelse": {
            "tekst": "Den ost er bare super smagfuld og dejlig indeni...",
            "smag": "mild",
            "vægt": 300,
            "billede": "stclemensblaaskimmel.jpg"
        }
    },
    {
        "varenr": 4767,
        "navn": "Fritos",
        "pris": 15,
        "type": "Cheddar",
        "producent": "Fritos",
        "antal": 76,
        "beskrivelse": {
            "tekst": "Mums... Dejlig ost",
            "smag": "mild",
            "vægt": 255,
            "billede": "cheddarmester.jpg"
        }
    },
    {
        "varenr": 5678,
        "navn": "Salatost i tern",
        "pris": 21.95,
        "type": "Salatost",
        "antal": 8,
        "producent": "Karolines køkken",
        "beskrivelse": {
            "vægt": 200,
            "smag": "mild",
            "billede": "fetatern.jpg",
            "tekst": "Letsaltet ost skåret i tern, lagt i et genluksbæger med si for nem håndtering."
        }
    },
    {
        "varenr": 420420,
        "navn": "buko pikant",
        "pris": 30,
        "type": "smøreost",
        "producent": "buko",
        "antal": 30,
        "beskrivelse": {
            "tekst": "God til de travle dage .",
            "smag": "blid",
            "vægt": 125,
            "billede": "smoereostpikant.jpg"
        }
    },
    {
        "varenr": 123,
        "navn": "Feta",
        "pris": 21.44,
        "producent": "Arla",
        "antal": 15,
        "beskrivelse": {
            "tekst": "Græsk feta er en hvid blød ost i saltlage. Den har en mild, men samtidig en tydelig saltsmag. Fetaosten er egnsbeskyttet, så det er kun feta produceret i Grækenland må kaldes Feta. Vi kalder derfor vores ost for Apetina®, når den sælges indenfor EU.",
            "smag": "Salt smag",
            "vægt": 1000,
            "billede": "Feta_hero.jpg"
        }
    },
    {
        "varenr": 666,
        "navn": "Red Devil",
        "pris": 666,
        "type": "chilli ost",
        "producent": "http://www.snowdoniacheese.co.uk/",
        "antal": 6666666,
        "beskrivelse": {
            "vægt": 2000,
            "smag": "stærk",
            "billede": "img/redDevilCheese.jpg",
            "tekst": "A judicious hit of chilli is the perfect foil for the rich savouriness of Red Leicester in this popular member of our range. As with all of our cheeses, Red Devil® is delicious enjoyed simply, but this one really comes into its own when melted over nachos, tortilla chips and barbecued burgers. It adds a depth of flavour and an intriguing heat."
        }
    },
    {
        "varenr": 5678,
        "navn": "Salatost i tern",
        "pris": 21.95,
        "type": "Salatost",
        "antal": 8,
        "producent": "Karolines køkken",
        "beskrivelse": {
            "vægt": 200,
            "smag": "mild",
            "billede": "fetatern.jpg",
            "tekst": "Letsaltet ost skåret i tern, lagt i et genluksbæger med si for nem håndtering."
        }
    },
    {
        "varenr": 6969,
        "navn": "Fuldmoden Gouda",
        "pris": 59.95,
        "type": "Gouda",
        "producent": "Gouda",
        "antal": 20,
        "beskrivelse": {
            "tekst": "Denne ost er bare for nice.",
            "smag": "Stærk",
            "vægt": 200,
            "billede": "goudaisthebest.jpg"
        }
    },
    {
        "varenr": 8259,
        "navn": "Thise Danbo",
        "pris": 35,
        "type": "Danbo",
        "antal": 20,
        "producent": "Thise",
        "beskrivelse": {
            "tekst": "Danbo er en dansk ost og den mest udbredte ostetype herhjemme.",
            "smag": "mild",
            "vægt": "530g",
            "billede": "thise.jpg"
        }
    },
    {
        "varenr": 8393,
        "navn": "Cornish Blue",
        "pris": 60,
        "type": "bluecheese",
        "producent": "Upton Cross.",
        "antal": 22,
        "beskrivelse": {
            "tekst": "Denne ost er cremede, mild og sød",
            "smag": "mild",
            "vægt": 500,
            "billede": "cornishbluecheese.jpg"
        }
    },
    {
        "varenr": 15,
        "navn": "Grøn Alpheost",
        "pris": 24.99,
        "type": "Smøreost",
        "producent": "Arla",
        "antal": "875",
        "beskrivelse": {
            "tekst": "Lavet fra en ko, klar til at smøre",
            "smag": "mild",
            "vægt": 250,
            "billede": "alpeost-iv.jpg"
        }
    },
    {
        "varenr": 7734,
        "navn": "Brians Delight",
        "pris": 116.99,
        "type": "Mascarpone",
        "producent": "Arla",
        "antal": 76,
        "beskrivelse": {
            "tekst": "Verdens bedste ost. Så er det sagt.",
            "smag": "voldstærk",
            "vægt": 20,
            "billede": "briansdelight.jpg"
        }
    },
    {
        "varenr": 2234,
        "navn": "Arla Klogborg",
        "pris": 69.5,
        "type": "lager",
        "producent": "Taulov Mejeri",
        "antal": 15,
        "beskrivelse": {
            "Tekst": "Den særlige Klovborg mælkesyrekultur er med til at give ostene den særlige lette, friske og runde ostesmag",
            "smag": "stærk",
            "vægt": 565,
            "billede": "stclemensblaaskimmel.jpg"
        }
    },
    {
        "varenr": 8729,
        "navn": "Myse",
        "pris": 20.95,
        "type": "Valle",
        "producent": "Myse",
        "antal": 90,
        "beskrivelse": {
            "tekst": "Valle, Myse ost er uovertruffen",
            "smag": "Mild",
            "vægt": 200,
            "billede": "myse.jpg"
        }
    },
    {
        "varenr": 22344,
        "navn": "mozzarella",
        "pris": 10,
        "typer": "pizza ost",
        "producent": "mozzarella",
        "antal": 300,
        "beskrivelse": {
            "Tekst": "God til pizza",
            "smag": "sød",
            "vægt": 100,
            "billede": "mozzeralla.png"
        }
    },
    {
        "varenr": 7654,
        "navn": "Cheesy",
        "pris": 6460,
        "typer": "hytteost",
        "producent": "Cheesy",
        "antal": 2345678,
        "beskrivelse": {
            "Tekst": "Mums til rugbrød",
            "smag": "cremet",
            "vægt": 500,
            "billede": "hytte.png"
        }
    },
    {
        "varenr": 5288,
        "navn": "Philadelphia",
        "pris": 29.5,
        "type": "Flødeeost",
        "producent": "Philadelphia",
        "antal": 49,
        "beskrivelse": {
            "Tekst": "Denne flødeost er mild og rundet i smagen og kan anbefales til brødprodukter.",
            "smag": "Mild",
            "vægt": 165,
            "billede": "philadelphia.jpg"
        }
    },
    {
        "varenr": 22344,
        "navn": "gouda",
        "pris": 15,
        "typer": "Inge ost",
        "producent": "Artla",
        "antal": 300,
        "beskrivelse": {
            "Tekst": "God til pizza",
            "smag": "gammel",
            "vægt": 100,
            "billede": "goudha.png"
        }
    },
    {
        "varenr": 55,
        "navn": "Parma ost",
        "pris": 50,
        "typer": "Parmasan",
        "producent": "Arla",
        "antal": 44,
        "beskrivelse": {
            "Tekst": "God til pasta",
            "smag": "tør",
            "vægt": 500,
            "billede": "Parmasan.png"
        }
    },
    {
        "varenr": 65,
        "navn": "rygeost",
        "pris": 13,
        "type": "rygeost",
        "producent": "Arla",
        "antal": 40,
        "beskrivelse": {
            "tekst": "Osten er en surmælksost, som saltes og krydres med kommen, og ryges med halm eller brændenælder",
            "smag": "Mild",
            "vægt": 20,
            "billede": "rygeost.jpg"
        }
    },
    {
        "varenr": 55,
        "navn": "Parma ost",
        "pris": 50,
        "typer": "Parmasan",
        "producent": "Arla",
        "antal": 44,
        "beskrivelse": {
            "Tekst": "God til pasta",
            "smag": "tør",
            "vægt": 500,
            "billede": "Parmasan.png"
        }
    },
    {
        "varenr": 6225,
        "navn": "ugga bugga - brie ost",
        "pris": 13,
        "type": "brie",
        "producent": "Arla",
        "antal": 40,
        "beskrivelse": {
            "tekst": "ad den er ikke lige alt for lækker",
            "smag": "Sur",
            "vægt": 20,
            "billede": "brie.jpg"
        }
    },
    {
        "varenr": 989875,
        "navn": "Nakke ost",
        "pris": 5,
        "typer": "Menneske snavs",
        "producent": "Tilfældig mand",
        "antal": 4,
        "beskrivelse": {
            "Tekst": "Kan ikke bruges til noget, men hvis det er din fetish, så lad gå da",
            "smag": "tør, men en smule cremet, kan være saltet",
            "vægt": 20,
            "billede": "18+nakkeost.png"
        }
    },
    {
        "varenr": 5415,
        "navn": "Havarti ost",
        "pris": 20,
        "type": "Havarti",
        "producent": "Nørup Mejeri",
        "antal": 41,
        "beskrivelse": {
            "tekst": "En mild ost med en rundet smag",
            "smag": "Mild",
            "vægt": 300,
            "billede": "havarti.jpg"
        }
    },
    {
        "varenr": 64234234,
        "navn": "Karamel sten",
        "pris": 59,
        "type": "Hvidskimmel",
        "producent": "Arla",
        "antal": 0,
        "beskrivelse": {
            "tekst": "Karamel Sten har gået under navnet ”Unika 22” de sidste tre år og som en del af vores Unika X sortiment været med på ostebilen, som besøger landets restauranter. Nu hvor Unika 22 er færdigudviklet har den som alle Unikas andre ostebørn fået et navn der symboliserer det særlige individ den er. Med smagen af karamel og en konsistens hård som en sten synes vi godt om navnet ”Karamel Sten”.",
            "smag": "Sur/salt",
            "vægt": 56,
            "billede": "brie.jpg"
        }
    }
];